### ISSUE TITLE
@PathVariable work with  Parameter Converter strangely

### ISSUE THREAD

**Issue by cxl086** [springfox/springfox#2053]
I use version 2.7.0 with Spring Boot 1.5.7. While I have code like below
```java
@RequestMapping(method = RequestMethod.GET,path = "/order/{orderId}/detail", produces = "application/json")
    @ApiOperation(value = "Detail of order.")
    public OrderDetail requireOrderManage(@AuthenticationPrincipal @ApiIgnore HelpersUser helpersUser,
                                          @PathVariable("orderId") Order order) {
        return orderService.detail(helpersUser.getUser(), order);
    }
```
and  define a Converter like this,
```java
@Component
public class StringToOrderConverter implements Converter<String, Order> {
    @Autowired
    OrderMapper orderMapper;

    @Override
    public Order convert(String source) {
        return orderMapper.selectByPrimaryKey(Integer.valueOf(source));
    }
}
```
Then the  document will like this
```json
"/api/v1/order/{orderId}/detail": {
            "get": {
                "consumes": [
                    "application/json"
                ],
                "operationId": "requireOrderManageUsingGET_1",
                "parameters": [
                    {
                        "format": "int32",
                        "in": "query",
                        "name": "address.id",
                        "required": false,
                        "type": "integer"
                    },
                    {
                        "description": "用户编号",
                        "format": "int32",
                        "in": "query",
                        "name": "address.userId",
                        "required": false,
                        "type": "integer"
                    },
                    {
                        "description": "联系人",
                        "in": "query",
                        "name": "address.contact",
                        "required": false,
                        "type": "string"
                    },
                    ...... // a ton of parameters
                    
                ],
                "produces": [
                    "application/json"
                ],
                "responses": {
                    "200": {
                        "description": "OK",
                        "schema": {
                            "$ref": "#/definitions/OrderDetail"
                        }
                    }
                },
                "security": [
                    {
                        "X-Auth-Token": []
                    }
                ],
                "summary": "Detail of Order",
                "tags": [
                    "order"
                ]
            }
        }
```
the generated document extract all properties of Order as request parameter.
Any way to make the document produce only "orderId" parameter like it does in v2.5.0 ?

**Comment by dilipkrish** [comment]
Thanks for reporting. This may be a bug and it may have worked accidentally in 2.5.0. Try the following as a work around...
- Mark the parameter with `@ApiIgnore` annotation
- Add an `@ApiImplicitParams` annotation for the orderId attribute.

**Comment by cxl086** [comment]
@ApiImplicitParams works. Is there any configuration of Docket can solve this problem?
thx for your work, anyway.

**Comment by dilipkrish** [comment]
@cxl086 you're welcome. No there isn't one using the docket unfortunately.

**Comment by dilipkrish** [comment]
Fixed via #2107
